.azure-pipelines/client.yml

```diff
@@ -87,6 +87,15 @@ jobs:
       vmImage: '$(OSVmImage)'
 
     steps:
+      - powershell:  'Invoke-WebRequest -Uri "$(VerifyAgentOSBlobPath)" -OutFile "./verify_agent_os.py"'
```

Comment:
Where is VerifyAgentOSBlobPath defined?

I'm kind of expecting you would be downloading a general zip containing a set of tools that we download and extract into a tools directory and then the next set of steps would just reference stuff from that tools directory.

Reply:
The URL is set as a Build Pipeline Variable. I see what you mean. So download all the sdk tools as a zip then extract that into a directory on the pipeline. I thought you didn't want us pulling in more tools than needed for the current job?

Reply:
>I thought you didn't want us pulling in more tools than needed for the current job?

In general I want to keep it minimal so we have to be careful of what we put into this set of tools but we need to balance the size of this toolset vs the complexity of manually downloading each tool/script individually. For now my goal is to have a single zip with a set of scripts we will be using in most of our definitions so that they can be bundled. Even if we don't use all the scripts in the bundle in a given build definition that should be OK but we should definitely be careful to not make the size too much.

Reply:
>The URL is set as a Build Pipeline Variable

I think we want the url to be in the yml file as part of the thing we need to ensure is that the exact version of the tool set is committed in the repo so we can reproduce the exact build if we need to go back and rebuild a older commit in the repo, which may be the case if we need to go back and hotfix and older build.